### PR TITLE
Feat: add enabled attribute for the configuration of UiElements

### PR DESCRIPTION
**Pull request by toofff**

### CODE REVIEW

**Comment by jacquesbh** [comment]
Hey!
Thank you for this!

It's a great idea.

I would like to discuss it a bit.

I think disabling a UI Element leads to many points:
- a disabled element should not be proposed in available UI Elements
- a disabled element should be parsed in admin, always, with a small warning indicating that the element is disabled, can be edited, but cannot be added anymore.
- a disabled element should not be displayed in front

In other words, if you disable an element a year later, it could be present in many contents. So we don't want to display it anymore (since it's disabled) but we don't want either to remove it from the old contents since I always say "do not remove data" (as a developer).

So, we need to continue this PR in order to make it work properly I think.

What do **you** think?

File: src/UiElement/Registry.php [hunk 1]
```diff
@@ -31,7 +31,7 @@ public function addUiElement(UiElementInterface $uiElement): void
         Assert::keyNotExists($this->uiElements, $uiElement->getCode(), 'UiElement with code "%s" is already registered.');
 
         $this->uiElements[$uiElement->getCode()] = $uiElement;
-        if (null !== $uiElement->getAlias()) {
+        if (null !== $uiElement->getAlias() && $uiElement->isEnabled()) {
```
Comment (jacquesbh) [inline src/UiElement/Registry.php:34]: I think we want to keep the alias here. It's important to be able to parse the elements in a JSON.
We just want to avoid displaying them, we do not want any error in old contents. I mean, we could re-enable the UI Element later…

File: src/UiElement/UiElementTrait.php [hunk 2]
```diff
@@ -64,6 +64,14 @@ public function getAlias(): ?string
         return $this->metadata->getAlias();
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function isEnabled(): bool
```
Comment (jacquesbh) [inline src/UiElement/UiElementTrait.php:70]: Where is this method used?
Is its purpose to be used inside the template? It could since we have access to this object in the template.

File: README.md [hunk 3]
```diff
@@ -175,6 +175,17 @@ The plugin already contains some simple elements.
 
 ![The Image collection element](screenshots/image_collection.png)
 
+### Deactivate an available element
+
+Define the overload of a proposed UiElement in your configuration folder, let's say in `config/packages/monsieurbiz_sylius_richeditor_plugin.yaml` as example.
```
Comment (jacquesbh) [inline README.md:180]: We could specify what happens to a disabled UI Element.

I would say:
- it's not displayed anymore.
- it's still editable in old contents but you can't add a new one.
- if the element has an alias, the alias is treated the same way.

**Comment by toofff** [comment]
@jacquesbh it was a first draft, but what you propose seems to me very good.

I'm watching this as soon as I can this week.

**Comment by kevinbalicot** [comment]
Hi @toofff !

Thanks for this PR, it's exactly what I need :D

Possible to check @jacquesbh comments soon ?

**Comment by toofff** [comment]
@kevinbalicot 

I thought I had time but his last days were very complicated. I will see to save myself time next week;)

**Review by maximehuran — APPROVED** [review]

**Comment by jacquesbh** [comment]
Hell yeah!

ping @kevinbalicot @toofff <3
